mcp/tools/jenkins.py

```
@@ -504,3 +506,109 @@ async def jenkins_whoami(
+        "BooleanParameterDefinition": "boolean",
+        "ChoiceParameterDefinition": "choice",
+        "TextParameterDefinition": "text",
+        "PasswordParameterDefinition": "password",
+        "FileParameterDefinition": "file",
+    }
+    return mapping.get(raw_type, "string")
+
+def _parse_config_xml(xml_text: str) -> List[dict]:
+    params = []
+    ns = {"jenkins": "http://hudson.model"}
+    root = ET.fromstring(xml_text)
```

Ruff (0.14.0)

Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)